### PR TITLE
feat: label_nodes removed

### LINKED ISSUE #169: Remove metadata
Remove metadata like `leaf`, ... . Deprecate label nodes

### PR DESCRIPTION
**Pull request by S4n60w3n**
Closes #169

### CODE REVIEW

File: license_sh/analyze/analyze_shared.py [hunk 1]
```diff
@@ -10,7 +10,8 @@
 from anytree import AnyNode, PreOrderIter
 
 from license_sh.analyze import lib
-from ..helpers import get_node_id
+
+NODE_ID_SEP = ":-:"
```
Comment (jvorcak) [inline license_sh/analyze/analyze_shared.py:14]: We should get rid of this separator and support native separators instead `==` or `@` instead + support pure functions. Or if we want to do it in another PR, please leave the TODO and corresponding issue.

File: tests/analyze/test_analyze_shared.py [hunk 2]
```diff
@@ -42,38 +42,42 @@ class Askalono_result:
...
+                AnyNode(name="child2", version="1.0.1", children=[AnyNode(name="child2Child", version="9.9.9")]),
             ],
         )
+        root_id = get_node_id("root", "1.2.3")
+        childChild_Id = get_node_id("childChild", "1.0.0")
         analyze = {
-            "root": [{"data": "Hey! I am a license text", "name": "Awesome license"}],
-            "childChild": [
+            root_id: [{"data": "Hey! I am a license text", "name": "Awesome license"}],
+            childChild_Id: [
```
Comment (jvorcak) [inline tests/analyze/test_analyze_shared.py:56]: `child_child_id`

File: license_sh/commands/run_license_sh.py [hunk 3]
```diff
@@ -81,8 +81,9 @@ def run_license_sh(arguments):
 
     dep_tree: PackageNode = run_check(project_to_check, path, silent, debug)
 
-    # TODO we should deprecate this
-    label_dep_tree(dep_tree, project_to_check)
+    dep_tree.data_version = __version__
+    dep_tree.project = project_to_check.value
```
Comment (jvorcak) [inline license_sh/commands/run_license_sh.py:85]: 👍

**Review by jvorcak — CHANGES_REQUESTED** [review]

File: license_sh/analyze/analyze_shared.py (hunk 1 above)
Comment (S4n60w3n) [reply license_sh/analyze/analyze_shared.py:14]: It is just used in 2 functions that are decoding and encoding name and version. It is nowhere saved

File: tests/analyze/test_analyze_shared.py (hunk 2 above)
Comment (S4n60w3n) [reply tests/analyze/test_analyze_shared.py:56]: well technically childChild is a name not a par of variable but ok :D

Comment (jvorcak) [reply tests/analyze/test_analyze_shared.py:56]: @S4n60w3n you're right, but it's a bit disturbing, :) but in general you're right, we can keep this

File: license_sh/analyze/analyze_shared.py (hunk 1 above)
Comment (jvorcak) [reply license_sh/analyze/analyze_shared.py:14]: @S4n60w3n if that is the case, then I'd consider using some other data structure. e.g. Tuple or NamedTuple.

Comment (S4n60w3n) [reply license_sh/analyze/analyze_shared.py]: Don't know what examply you mean but i have updated it a bit

**Review by jvorcak — APPROVED** [review]
Much better 👏
